src/scn/BlockchainManager/BlockchainManager.h

```diff
@@ -42,22 +42,22 @@ namespace scn {
 
     class BlockchainManager {
     public:
-        BlockchainManager(public_key_t our_public_key,
-                private_key_t our_private_key,
+        BlockchainManager(const public_key_t& our_public_key,
+                const private_key_t& our_private_key,
                 Blockchain& blockchain,
                 IP2PConnector& p2p_connector,
                 IMiner& miner,
                 bool initial_fetch = true,
                 ISynchronizedTimer& sync_timer = static_sync_timer);
         virtual ~BlockchainManager();
 
-        virtual void baselineBlockReceivedCallback(const peer_id_t& peer_id, std::shared_ptr<const BaselineBlock> block, bool reply);
+        virtual void baselineBlockReceivedCallback(const peer_id_t& peer_id, const std::shared_ptr<const BaselineBlock>& block, bool reply);
 
-        virtual void collectionBlockReceivedCallback(const peer_id_t& peer_id, std::shared_ptr<const CollectionBlock> block, bool reply);
+        virtual void collectionBlockReceivedCallback(const peer_id_t& peer_id, const std::shared_ptr<const CollectionBlock>& block, bool reply);
 
```

Comment:
I don't really see the sense of passing a const reference to the shared_ptr here (except minimal performance gains).
I personally prefer here to pass the shared_ptr by value to allow the callback to process the blocks asynchronously (the blocks may get pretty big).
See here: https://www.modernescpp.com/index.php/c-core-guidelines-passing-smart-pointer